// @tamagui/themes/dist/cjs/componentThemeDefinitions.native.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var componentThemeDefinitions_exports = {};
__export(componentThemeDefinitions_exports, {
  componentThemeDefinitions: () => componentThemeDefinitions,
  overlayThemeDefinitions: () => overlayThemeDefinitions
});
module.exports = __toCommonJS(componentThemeDefinitions_exports);
var import_templates = require("./templates");
function _define_property(obj, key, value) {
  return key in obj ? Object.defineProperty(obj, key, {
    value,
    enumerable: !0,
    configurable: !0,
    writable: !0
  }) : obj[key] = value, obj;
}
function _object_spread(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {}, ownKeys = Object.keys(source);
    typeof Object.getOwnPropertySymbols == "function" && (ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
      return Object.getOwnPropertyDescriptor(source, sym).enumerable;
    }))), ownKeys.forEach(function(key) {
      _define_property(target, key, source[key]);
    });
  }
  return target;
}
var overlayThemes = {
  light: {
    background: "rgba(0,0,0,0.5)"
  },
  dark: {
    background: "rgba(0,0,0,0.9)"
  }
}, overlayThemeDefinitions = [
  {
    parent: "light",
    theme: overlayThemes.light
  },
  {
    parent: "dark",
    theme: overlayThemes.dark
  }
], componentThemeDefinitions = {
  ListItem: [
    _object_spread({
      parent: "light",
      mask: "strengthen"
    }, import_templates.maskOptions.component),
    _object_spread({
      parent: "dark",
      mask: "identity"
    }, import_templates.maskOptions.component)
  ],
  Card: _object_spread({
    mask: "soften"
  }, import_templates.maskOptions.component),
  Button: _object_spread({
    mask: "soften2"
  }, import_templates.maskOptions.button),
  Checkbox: _object_spread({
    mask: "softenBorder2"
  }, import_templates.maskOptions.component),
  Switch: _object_spread({
    mask: "soften2"
  }, import_templates.maskOptions.component),
  SwitchThumb: _object_spread({
    mask: "inverseStrengthen2"
  }, import_templates.maskOptions.component),
  TooltipContent: _object_spread({
    mask: "soften2"
  }, import_templates.maskOptions.component),
  DrawerFrame: _object_spread({
    mask: "soften"
  }, import_templates.maskOptions.component),
  Progress: _object_spread({
    mask: "soften"
  }, import_templates.maskOptions.component),
  RadioGroupItem: _object_spread({
    mask: "softenBorder2"
  }, import_templates.maskOptions.component),
  TooltipArrow: _object_spread({
    mask: "soften"
  }, import_templates.maskOptions.component),
  SliderTrackActive: _object_spread({
    mask: "inverseSoften"
  }, import_templates.maskOptions.component),
  SliderTrack: _object_spread({
    mask: "soften2"
  }, import_templates.maskOptions.component),
  SliderThumb: _object_spread({
    mask: "inverse"
  }, import_templates.maskOptions.component),
  Tooltip: _object_spread({
    mask: "inverse"
  }, import_templates.maskOptions.component),
  ProgressIndicator: _object_spread({
    mask: "inverse"
  }, import_templates.maskOptions.component),
  SheetOverlay: overlayThemeDefinitions,
  DialogOverlay: overlayThemeDefinitions,
  ModalOverlay: overlayThemeDefinitions,
  Input: _object_spread({
    mask: "softenBorder2"
  }, import_templates.maskOptions.component),
  TextArea: _object_spread({
    mask: "softenBorder2"
  }, import_templates.maskOptions.component)
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  componentThemeDefinitions,
  overlayThemeDefinitions
});
