// @tamagui/themes/dist/cjs/themes-new.native.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var themes_new_exports = {};
__export(themes_new_exports, {
  themes: () => themes
});
module.exports = __toCommonJS(themes_new_exports);
var import_theme_builder = require("@tamagui/theme-builder"), import_componentThemeDefinitions = require("./componentThemeDefinitions"), import_masks = require("./masks"), import_palettes = require("./palettes"), import_shadows = require("./shadows"), import_templates = require("./templates"), import_tokens = require("./tokens");
function _define_property(obj, key, value) {
  return key in obj ? Object.defineProperty(obj, key, {
    value,
    enumerable: !0,
    configurable: !0,
    writable: !0
  }) : obj[key] = value, obj;
}
function _object_spread(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {}, ownKeys = Object.keys(source);
    typeof Object.getOwnPropertySymbols == "function" && (ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
      return Object.getOwnPropertyDescriptor(source, sym).enumerable;
    }))), ownKeys.forEach(function(key) {
      _define_property(target, key, source[key]);
    });
  }
  return target;
}
var colorThemeDefinition = function(colorName) {
  return [
    {
      parent: "light",
      palette: colorName,
      template: "colorLight"
    },
    {
      parent: "dark",
      palette: colorName,
      template: "base"
    }
  ];
}, themesBuilder = (0, import_theme_builder.createThemeBuilder)().addPalettes(import_palettes.palettes).addTemplates(import_templates.templates).addMasks(import_masks.masks).addThemes({
  light: {
    template: "base",
    palette: "light",
    nonInheritedValues: _object_spread({}, import_tokens.lightColors, import_shadows.shadows.light)
  },
  dark: {
    template: "base",
    palette: "dark",
    nonInheritedValues: _object_spread({}, import_tokens.darkColors, import_shadows.shadows.dark)
  }
}).addChildThemes({
  orange: colorThemeDefinition("orange"),
  yellow: colorThemeDefinition("yellow"),
  green: colorThemeDefinition("green"),
  blue: colorThemeDefinition("blue"),
  purple: colorThemeDefinition("purple"),
  pink: colorThemeDefinition("pink"),
  red: colorThemeDefinition("red")
}).addChildThemes({
  alt1: _object_spread({
    mask: "soften"
  }, import_templates.maskOptions.alt),
  alt2: _object_spread({
    mask: "soften2"
  }, import_templates.maskOptions.alt),
  active: {
    mask: "soften3",
    skip: {
      color: 1
    }
  }
}).addChildThemes(import_componentThemeDefinitions.componentThemeDefinitions, {}), themes = themesBuilder.build();
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  themes
});
