// @tamagui/animations-moti/dist/cjs/createAnimations.native.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf, __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: !0 }) : target,
  mod
)), __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var createAnimations_exports = {};
__export(createAnimations_exports, {
  createAnimations: () => createAnimations
});
module.exports = __toCommonJS(createAnimations_exports);
var import_use_presence = require("@tamagui/use-presence"), import_author = require("moti/author"), import_react = require("react"), import_react_native_reanimated = __toESM(require("react-native-reanimated"));
function _define_property(obj, key, value) {
  return key in obj ? Object.defineProperty(obj, key, {
    value,
    enumerable: !0,
    configurable: !0,
    writable: !0
  }) : obj[key] = value, obj;
}
function _object_spread(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {}, ownKeys = Object.keys(source);
    typeof Object.getOwnPropertySymbols == "function" && (ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
      return Object.getOwnPropertyDescriptor(source, sym).enumerable;
    }))), ownKeys.forEach(function(key) {
      _define_property(target, key, source[key]);
    });
  }
  return target;
}
var onlyAnimateKeys = {
  transform: !0,
  opacity: !0,
  height: !0,
  width: !0,
  backgroundColor: !0,
  borderColor: !0,
  borderLeftColor: !0,
  borderRightColor: !0,
  borderTopColor: !0,
  borderBottomColor: !0,
  borderRadius: !0,
  borderTopLeftRadius: !0,
  borderTopRightRadius: !0,
  borderBottomLeftRadius: !0,
  borderBottomRightRadius: !0,
  borderLeftWidth: !0,
  borderRightWidth: !0,
  borderTopWidth: !0,
  borderBottomWidth: !0,
  color: !0,
  left: !0,
  right: !0,
  top: !0,
  bottom: !0
};
function createAnimations(animations) {
  return {
    // View: isWeb ? AnimatedView : Animated.View,
    // Text: isWeb ? AnimatedText : Animated.Text,
    View: import_react_native_reanimated.default.View,
    Text: import_react_native_reanimated.default.Text,
    isReactNative: !0,
    animations,
    usePresence: import_use_presence.usePresence,
    ResetPresence: import_use_presence.ResetPresence,
    useAnimatedNumber: function(initial) {
      var sharedValue = (0, import_react_native_reanimated.useSharedValue)(initial);
      return (0, import_react.useMemo)(function() {
        return {
          getInstance: function() {
            "worklet";
            return sharedValue;
          },
          getValue: function() {
            "worklet";
            return sharedValue.value;
          },
          setValue: function(next) {
            "worklet";
            var config = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {
              type: "spring"
            }, onFinish = arguments.length > 2 ? arguments[2] : void 0;
            config.type === "direct" ? (sharedValue.value = next, onFinish == null || onFinish()) : config.type === "spring" ? sharedValue.value = (0, import_react_native_reanimated.withSpring)(next, config, onFinish ? function() {
              "worklet";
              (0, import_react_native_reanimated.runOnJS)(onFinish)();
            } : void 0) : sharedValue.value = (0, import_react_native_reanimated.withTiming)(next, config, onFinish ? function() {
              "worklet";
              (0, import_react_native_reanimated.runOnJS)(onFinish)();
            } : void 0);
          },
          stop: function() {
            "worklet";
            (0, import_react_native_reanimated.cancelAnimation)(sharedValue);
          }
        };
      }, [
        sharedValue
      ]);
    },
    useAnimatedNumberReaction: function(param, onValue) {
      var value = param.value, instance = value.getInstance();
      return (0, import_react_native_reanimated.useAnimatedReaction)(
        function() {
          return instance.value;
        },
        function(next, prev) {
          prev !== next && (0, import_react_native_reanimated.runOnJS)(onValue)(next);
        },
        // dependency array is very important here
        [
          onValue,
          instance
        ]
      );
    },
    /**
    * `getStyle` must be a worklet
    */
    useAnimatedNumberStyle: function(val, getStyle) {
      var instance = val.getInstance(), derivedValue = (0, import_react_native_reanimated.useDerivedValue)(
        function() {
          return instance.value;
        },
        // dependency array is very important here
        [
          instance,
          getStyle
        ]
      );
      return (0, import_react_native_reanimated.useAnimatedStyle)(
        function() {
          return getStyle(derivedValue.value);
        },
        // dependency array is very important here
        [
          val,
          getStyle,
          derivedValue,
          instance
        ]
      );
    },
    useAnimations: function(animationProps) {
      var props = animationProps.props, presence = animationProps.presence, style = animationProps.style, onDidAnimate = animationProps.onDidAnimate, componentState = animationProps.componentState, animationKey = Array.isArray(props.animation) ? props.animation[0] : props.animation, isHydrating = componentState.unmounted === "should-enter", animate = {}, dontAnimate = {};
      if (isHydrating)
        dontAnimate = style;
      else {
        var animateOnly = props.animateOnly;
        for (var key in style) {
          var value = style[key];
          !onlyAnimateKeys[key] || value === "auto" || animateOnly && !animateOnly.includes(key) ? dontAnimate[key] = value : animate[key] = value;
        }
      }
      var animateStr = JSON.stringify(animate), styles = (0, import_react.useMemo)(function() {
        return JSON.parse(animateStr);
      }, [
        animateStr
      ]), isExiting = !!(presence != null && presence[1]), presenceContext = (0, import_react.useContext)(import_use_presence.PresenceContext), usePresenceValue = presence || void 0, transition = _object_spread({}, animations[animationKey]);
      if (Array.isArray(props.animation)) {
        var config = props.animation[1];
        if (config && typeof config == "object")
          for (var key1 in config) {
            var val = config[key1];
            typeof val == "string" ? transition[key1] = animations[val] : transition[key1] = val;
          }
      }
      var motiProps = {
        animate: isExiting || isHydrating ? {} : styles,
        transition,
        usePresenceValue,
        presenceContext,
        exit: isExiting ? styles : void 0
      }, moti = (0, import_author.useMotify)(motiProps);
      return process.env.NODE_ENV === "development" && props.debug && console.info("useMotify(", JSON.stringify(motiProps, null, 2) + ")", {
        animationProps,
        motiProps,
        moti,
        style: [
          dontAnimate,
          moti.style
        ]
      }), {
        style: [
          dontAnimate,
          moti.style
        ]
      };
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createAnimations
});
