// @tamagui/themes/dist/esm/componentThemeDefinitions.native.js
import { maskOptions } from "./templates";
function _define_property(obj, key, value) {
  return key in obj ? Object.defineProperty(obj, key, {
    value,
    enumerable: !0,
    configurable: !0,
    writable: !0
  }) : obj[key] = value, obj;
}
function _object_spread(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {}, ownKeys = Object.keys(source);
    typeof Object.getOwnPropertySymbols == "function" && (ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
      return Object.getOwnPropertyDescriptor(source, sym).enumerable;
    }))), ownKeys.forEach(function(key) {
      _define_property(target, key, source[key]);
    });
  }
  return target;
}
var overlayThemes = {
  light: {
    background: "rgba(0,0,0,0.5)"
  },
  dark: {
    background: "rgba(0,0,0,0.9)"
  }
}, overlayThemeDefinitions = [
  {
    parent: "light",
    theme: overlayThemes.light
  },
  {
    parent: "dark",
    theme: overlayThemes.dark
  }
], componentThemeDefinitions = {
  ListItem: [
    _object_spread({
      parent: "light",
      mask: "strengthen"
    }, maskOptions.component),
    _object_spread({
      parent: "dark",
      mask: "identity"
    }, maskOptions.component)
  ],
  Card: _object_spread({
    mask: "soften"
  }, maskOptions.component),
  Button: _object_spread({
    mask: "soften2"
  }, maskOptions.button),
  Checkbox: _object_spread({
    mask: "softenBorder2"
  }, maskOptions.component),
  Switch: _object_spread({
    mask: "soften2"
  }, maskOptions.component),
  SwitchThumb: _object_spread({
    mask: "inverseStrengthen2"
  }, maskOptions.component),
  TooltipContent: _object_spread({
    mask: "soften2"
  }, maskOptions.component),
  DrawerFrame: _object_spread({
    mask: "soften"
  }, maskOptions.component),
  Progress: _object_spread({
    mask: "soften"
  }, maskOptions.component),
  RadioGroupItem: _object_spread({
    mask: "softenBorder2"
  }, maskOptions.component),
  TooltipArrow: _object_spread({
    mask: "soften"
  }, maskOptions.component),
  SliderTrackActive: _object_spread({
    mask: "inverseSoften"
  }, maskOptions.component),
  SliderTrack: _object_spread({
    mask: "soften2"
  }, maskOptions.component),
  SliderThumb: _object_spread({
    mask: "inverse"
  }, maskOptions.component),
  Tooltip: _object_spread({
    mask: "inverse"
  }, maskOptions.component),
  ProgressIndicator: _object_spread({
    mask: "inverse"
  }, maskOptions.component),
  SheetOverlay: overlayThemeDefinitions,
  DialogOverlay: overlayThemeDefinitions,
  ModalOverlay: overlayThemeDefinitions,
  Input: _object_spread({
    mask: "softenBorder2"
  }, maskOptions.component),
  TextArea: _object_spread({
    mask: "softenBorder2"
  }, maskOptions.component)
};
export {
  componentThemeDefinitions,
  overlayThemeDefinitions
};
